ADAL/src/telemetry/ADTelemetryDefaultEvent.m

```diff
@@ -125,6 +125,19 @@ - (NSString*)getStringFromDate:(NSDate*)date
     return [s_dateFormatter stringFromDate:date];
 }
 
+- (void)sanitizePropertyNames
+{
+    NSMutableDictionary* propertyMapCopy = [_propertyMap copy];
+    [_propertyMap removeAllObjects];
+    
+    for (NSString* propertyName in propertyMapCopy)
+    {
+        // Some telemetry collectors only allow alphabetic characters, underscores, and dots in the property names
+        [_propertyMap setObject:[propertyMapCopy objectForKey:propertyName]
+                         forKey:[propertyName stringByReplacingOccurrencesOfString:@"-" withString:@"_"]];
```

Comment:
Checked with @jasoncoolmax before I made this change, these keys are not only used by telemetry, Changing these values will break existing apps.